packages/dd-trace/src/config.js

```diff
@@ -375,6 +378,10 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
       process.env.DD_TELEMETRY_LOG_COLLECTION_ENABLED,
       DD_IAST_ENABLED
     )
+    const DD_TELEMETRY_DIAGNOSTIC_LOG_COLLECTION_ENABLED = coalesce(
```

Comment:
Why is this no longer enabled by default? How will we know if there is an issue?

Reply:
Also, using a different term here could be confusing. Why "diagnostic" instead of "debug" like what is used elsewhere?

Reply:
This differs a lot from our current approach. Why not use `DEBUG` and `LOG_LEVEL` to be consistent? We'll probably want to normalize all of this properly across the library later on too.

Reply:
I'm going to remove `DD_TELEMETRY_DIAGNOSTIC_LOG_COLLECTION_ENABLED` for now because it has no effect. And I'll try to talk with the RFC creators.

Reply:
Ok, we've decided to remove it until we are interested in collecting DEBUG or INFO traces.

But `DD_TELEMETRY_DIAGNOSTIC_LOG_COLLECTION_ENABLED` has other implications besides changing the log level. If `true`:
- logs with stack traces sent via telemetry are not redacted and therefore no sensitive information is removed from them.
- and it is supposed to be valid for 4 hours (to make sure that customers don’t forget)

Reply:
that's super weird, and I agree with roch :/

Reply:
it has been removed from our implementation but I suppose java and other teams are using it.
wdyt @uurien @rochdev?

Reply:
Yeah let's make sure it makes sense _before_ implementing it.